lib/auth/api.go

```diff
@@ -721,6 +721,76 @@ type DiscoveryAccessPoint interface {
 	DeleteDatabase(ctx context.Context, name string) error
 }
 
+// ReadOktaAccessPoint is a read only API interface to be
+// used by an Okta component.
+//
+// NOTE: This interface must match the resources replicated in cache.ForOkta except for
+// access requests, which are expected to only be used by events.
+type ReadOktaAccessPoint interface {
+	// Closer closes all the resources
+	io.Closer
+
+	// NewWatcher returns a new event watcher.
+	NewWatcher(ctx context.Context, watch types.Watch) (types.Watcher, error)
+
+	// GetUser returns a services.User for this cluster.
+	GetUser(name string, withSecrets bool) (types.User, error)
+
+	// ListUserGroups returns a paginated list of all user group resources.
+	ListUserGroups(context.Context, int, string) ([]types.UserGroup, string, error)
+
+	// GetUserGroup returns the specified user group resources.
+	GetUserGroup(ctx context.Context, name string) (types.UserGroup, error)
+
+	// ListOktaImportRules returns a paginated list of all Okta import rule resources.
+	ListOktaImportRules(context.Context, int, string) ([]types.OktaImportRule, string, error)
+
+	// GetOktaImportRule returns the specified Okta import rule resources.
+	GetOktaImportRule(ctx context.Context, name string) (types.OktaImportRule, error)
+
+	// ListOktaAssignments returns a paginated list of all Okta assignment resources.
+	ListOktaAssignments(context.Context, int, string) ([]types.OktaAssignment, string, error)
+
+	// GetOktaAssignmen treturns the specified Okta assignment resources.
+	GetOktaAssignment(ctx context.Context, name string) (types.OktaAssignment, error)
+}
+
+// OktaAccessPoint is a read only cache interface used by an Okta component.
```

Comment:
It doesn't look like it's read only

Reply:
Ahh whoops, I got a clarification after writing this comment and forgot to update it. Thank you!